File: app/src/main/java/com/goranch/publicapis/ui/webview/WebFragment.java
package com.goranch.publicapis.ui.webview;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.WebChromeClient;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.ProgressBar;

import com.goranch.publicapis.R;
import com.goranch.publicapis.ui.details.DetailsShazamFragment;
import com.goranch.publicapis.ui.food.fragment.DetailsFragment;

import butterknife.Bind;
import butterknife.ButterKnife;

public class WebFragment extends Fragment {

    @Bind(R.id.web_view)
    WebView vw;

    @Bind(R.id.progressBar)
    ProgressBar progressBar;

    String contentUrl;

    public static WebFragment newInstance() {
        return new WebFragment();
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            if (getArguments().containsKey(DetailsShazamFragment.URL)) {
                contentUrl = getArguments().getString(DetailsShazamFragment.URL);
            } else if (getArguments().containsKey(DetailsFragment.URL)) {
                contentUrl = getArguments().getString(DetailsFragment.URL);
            }
        }
    }



    @SuppressLint("SetJavaScriptEnabled")
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        View v = inflater.inflate(R.layout.web_view_fragment, container, false);

        ButterKnife.bind(this, v);

        vw.getSettings().setJavaScriptEnabled(true);

        vw.setWebViewClient(new WebViewClient() {

            @Override
            public void onPageFinished(WebView view, String url) {
                super.onPageFinished(view, url);
                progressBar.setVisibility(View.GONE);
            }
        });

        vw.setWebChromeClient(new WebChromeClient() {
            public void onProgressChanged(WebView view, int progress) {

                progressBar.setProgress(progress);
                if (progress == 100)
                    //Make the bar disappear after URL is loaded
                    progressBar.setVisibility(View.GONE);
            }
        });

        if (contentUrl != null) {
            vw.loadUrl(contentUrl);
        }

        return v;
    }
}

File: app/src/main/java/com/goranch/publicapis/ui/webview/WebFragment.kt
package com.goranch.publicapis.ui.webview

import android.annotation.SuppressLint
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.webkit.WebChromeClient
import android.webkit.WebView
import android.webkit.WebViewClient
import android.widget.ProgressBar
import butterknife.BindView
import butterknife.ButterKnife
import com.goranch.publicapis.R
import com.goranch.publicapis.ui.food.fragment.DetailsFragment

class WebFragment : Fragment() {

    @JvmField
    @BindView(R.id.web_view)
    internal var vw: WebView? = null

    @JvmField
    @BindView(R.id.progressBar)
    internal var progressBar: ProgressBar? = null

    internal var contentUrl: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            if (arguments.containsKey(DetailsFragment.URL)) {
                contentUrl = arguments.getString(DetailsFragment.URL)
            }
        }
    }


    @SuppressLint("SetJavaScriptEnabled")
    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {

        val v = inflater!!.inflate(R.layout.web_view_fragment, container, false)

        ButterKnife.bind(this, v)

        vw!!.settings.javaScriptEnabled = true

        vw!!.webViewClient = object : WebViewClient() {

            override fun onPageFinished(view: WebView, url: String) {
                super.onPageFinished(view, url)
                progressBar!!.visibility = View.GONE
            }
        }

        vw!!.webChromeClient = object : WebChromeClient() {
            override fun onProgressChanged(view: WebView, progress: Int) {

                progressBar!!.progress = progress
                if (progress == 100)
                //Make the bar disappear after URL is loaded
                    progressBar!!.visibility = View.GONE
            }
        }

        if (contentUrl != null) {
            vw!!.loadUrl(contentUrl)
        }

        return v
    }

    companion object {

        fun newInstance(url: String): WebFragment {
            val b = Bundle()
            b.putString(DetailsFragment.URL, url)

            val fragment = WebFragment()
            fragment.arguments = b
            return fragment
        }
    }
}
